internal/internal_task_pollers.go

```diff
@@ -279,15 +283,17 @@ func newWorkflowTaskPoller(
 	params workerExecutionParameters,
 ) *workflowTaskPoller {
 	return &workflowTaskPoller{
-		basePoller:                   basePoller{shutdownC: params.WorkerStopChannel},
+		basePoller: basePoller{
+			shutdownC:    params.WorkerStopChannel,
+			metricsScope: metrics.NewTaggedScope(params.MetricsScope),
+			logger:       params.Logger,
```

Comment:
might be worth adding a tag (to both) to make it clear which thing is panicking... though the stack trace will probably make that clear too.

Reply:
Added, good call